api/objects.proto

```diff
@@ -222,3 +222,35 @@ message Cluster {
 	// a new key is allocated on key rotation.
 	uint64 encryption_key_lamport_clock = 6;
 }
+
+// Secret represents a secret that should be passed to a container or a node,
+// and is immutable. It wraps the `spec` provided by the user with useful
+// information that is generated from the secret data in the `spec`, such as
+// the digest and size of the secret data.
+message Secret {
```

Comment:
In this new model without `SecretValue`, don't you need a way to check which secret with a given name is the newest? I think there may need to be some kind of versioning field (to check which secret is most current, as opposed to the version in Meta, which tracks the last modification to the object). The simplest thing I can think of is an integer on the `Secret` object. When we create a new secret, we'd list all secrets with the same name and set that counter to `max(versions) + 1`.